superset/assets/src/explore/components/controls/VizTypeControl.jsx

```diff
@@ -89,13 +96,42 @@ export default class VizTypeControl extends React.PureComponent {
         </div>
       </div>);
   }
+  getVizTypeByKey(types, key) {
```

Comment:
Better create an object or map for lookup. It will be more efficient. This is currently `O(n)` for every call.